tests/c/csound_orc_semantics_test.cpp

```diff
@@ -0,0 +1,242 @@
+ *
+ * Created on June 7, 2012, 4:03 PM
+ */
+
+#define __BUILDING_LIBCSOUND
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "csoundCore.h"
+#include "csound_orc.h"
+#include "gtest/gtest.h"
+
+extern "C" {
+    extern OENTRIES* find_opcode2 (CSOUND* csound, char* opname);
+    extern OENTRY* resolve_opcode (CSOUND*, OENTRIES* entries, char* outArgTypes, char* inArgTypes);
+    extern OENTRY* find_opcode_new (CSOUND* csound, char* opname, char* outArgsFound, char* inArgsFound);
+
+    extern bool check_in_arg (char* found, char* required);
+    extern bool check_in_args (CSOUND* csound, char* outArgsFound, char* opOutArgs);
+    extern bool check_out_arg (char* found, char* required);
+    extern bool check_out_args (CSOUND* csound, char* outArgsFound, char* opOutArgs);
+}
+
+class OrcSemanticsTest : public ::testing::Test {
+public:
+    OrcSemanticsTest ()
+    {
+    }
+
+    virtual ~OrcSemanticsTest ()
+    {
+    }
+
+    virtual void SetUp ()
+    {
+        csoundSetGlobalEnv ("OPCODE6DIR64", "../../");
+        csound = csoundCreate (0);
+        csoundCreateMessageBuffer (csound, 0);
+        csoundSetOption (csound, "--logfile=NULL");
+    }
+
+    virtual void TearDown ()
+    {
+        csoundCleanup (csound);
+        csoundDestroyMessageBuffer (csound);
+        csoundDestroy (csound);
+        csound = nullptr;
+    }
+
+    CSOUND* csound {nullptr};
+};
+
+TEST_F (OrcSemanticsTest, FindOpcode2Test)
+{
+    OENTRIES* entries = find_opcode2(csound, "=");
+    entries = find_opcode2(csound, "vco2");
+    ASSERT_EQ (1, entries->count);
+    csound->Free(csound, entries);
+}
+
+TEST_F (OrcSemanticsTest, ResolveOpcodeTest)
+{
+    OENTRIES* entries = find_opcode2(csound, "=");
+    
+    OENTRY* opc = resolve_opcode(csound, entries, "k", "k");
+    ASSERT_TRUE (opc != NULL);
+    csound->Free(csound, entries);
+    
+    entries = find_opcode2(csound, "vco2");
+    ASSERT_EQ (1, entries->count);
+    
+    opc = resolve_opcode(csound, entries, "a", "cc");
+    ASSERT_TRUE (opc != NULL);
+    csound->Free(csound, entries);
+    
+    entries = find_opcode2(csound, "passign");
+    ASSERT_EQ (3, entries->count);
+
+    opc = resolve_opcode(csound, entries, "iiiiS", NULL);
+    ASSERT_TRUE (opc != NULL);
+    csound->Free(csound, entries);
+    
+    entries = find_opcode2(csound, "pcauchy");
+    opc = resolve_opcode(csound, entries, "i", "k");
+    ASSERT_TRUE (opc->iopadr != NULL);
+    
+    opc = resolve_opcode(csound, entries, "k", "k");
+    ASSERT_TRUE (opc->kopadr != NULL);
+    
+    opc = resolve_opcode(csound, entries, "a", "k");
+    ASSERT_TRUE (opc->aopadr != NULL);
+    
+    csound->Free(csound, entries);
+}
+
+TEST_F (OrcSemanticsTest, FindOpcodeNewTest)
+{
+    ASSERT_TRUE (find_opcode_new(csound, "##error", "i", "i") != NULL);
+    ASSERT_TRUE (find_opcode_new(csound, "##error", NULL, "i") == NULL);
+    ASSERT_TRUE (find_opcode_new(csound, "##xin64", "i", NULL) != NULL);
```

Comment:
This assertion fails as well. Can't find xin64
```
G:\Programming\GitHub\csound\tests\c\csound_orc_semantics_test.cpp(104): error: Value of: find_opcode_new(csound, "##xin64", "i", NULL) != NULL
  Actual: false
Expected: true
```

Reply:
check out commit 194ad034519296a1472ec4ad773ccc8f208c9d78 looks like xin64 was removed.

Reply:
Is it removed or renamed xin.64? Can just delete that assertion I guess.